library/common/api/external.cc

```diff
@@ -0,0 +1,25 @@
+#include "external.h"
+
+#include <unordered_map>
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Api {
+namespace External {
+
+static std::unordered_map<std::string, void*> registry_{};
```

Comment:
How will this work with instance-based engines? Will the registry be per instance, or do you see this always being static?

Reply:
Per-instance. I'm going to do a pass and flag and link anything that needs to be updated once the engine is instantiated.